metal/drivers/sifive_pio0.h

```diff
@@ -0,0 +1,29 @@
+#include <stdlib.h>
+#include <stdint.h>
+#include <metal/compiler.h>
+
+#define PIO_BASE 0x60000
+
+struct metal_pio;
+
+struct metal_pio_vtable {
+    void     (*v_pio_write)(uint32_t *pio_base, uint32_t data, uint32_t enable);
+    uint32_t (*v_pio_read)(uint32_t *pio_base);
+};
+
+struct metal_pio {
+    uint32_t *pio_base;
+    const struct metal_pio_vtable vtable;
+};
+
+__METAL_DECLARE_VTABLE(metal_pio);
+
+extern const struct metal_pio * pio_tables[];
+extern uint8_t pio_tables_cnt;
```

Comment:
These two variables don't need to be declared publicly. I'd also like to not have these at all, since we should be reading this information from DTS in the first place, but if you are going to have this in here in the meantime, then please don't unnecessarily export variables.

Reply:
fixed